runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -631,7 +631,12 @@ TR::SymbolValidationManager::addClassFromCPRecord(TR_OpaqueClassBlock *clazz, J9
    if (recordExists(&byName))
       return true; // already have an equivalent ClassByName
 
-   bool added = addClassRecord(clazz, new (_region) ClassFromCPRecord(clazz, beholder, cpIndex));
+   bool added;
+   if (!isAlreadyValidated(clazz)) // save a ClassChainRecord
```

Comment:
The idea here is that 
1. because a ClassFromCP+ClassChain is larger than ClassByName, it is better to generate a ClassByName if we've never generated a ClassChain record. 
2. because ClassFromCP is smaller than ClassByName, it is better to generate a ClassFromCP if we **have** generated a ClassChain record.

Is that right?

Reply:
Yes, especially since in 2 the ClassFromCP will suppress any later ClassByName that would be equivalent